Fixes to use dygraphs 1.1

chrisirhc opened the pull request:
- Add default `xAxisLabelFormatter` when no xAxisLabelFormatter is specified
- Default to display MMM YY for monthly x axis labels
- Allow using legend `show="follow"`
- Move axisLabelWidth option to axes options

<img width="775" alt="screen shot 2015-10-10 at 3 04 51 pm" src="https://cloud.githubusercontent.com/assets/132584/10413511/764f3abc-6f60-11e5-9376-5f7ffdc389a0.png">

chrisirhc commented:
Just saw that #84 has some overlap with this with regards to deprecated options. Those changes aren't important to this PR so I'm happy to remove / rebase them for this PR.

jjallaire commented:
Unfortunately the change to 60 pixels is not enough to prevent a huge decrease in the number of labels on the y-axis in some cases (e.g. try the example code here with the change: http://rstudio.github.io/dygraphs/gallery-axis-options.html).

jjallaire commented on R/options.R:
```diff
@@ -206,8 +200,6 @@ dyOptions <- function(dygraph,
   options$colorValue <- colorValue
   options$colorSaturation <- colorSaturation
-  options$drawXAxis <- drawXAxis
-  options$drawYAxis <- drawYAxis
```
Here we're breaking existing code that passes those values. The values need to be set to the correct spot within the individual axis options (or you could drop this code in favor of the other PR).

chrisirhc replied:
Yea, I'll drop these changes instead since they're not required for this PR.

chrisirhc commented:
Ah, okay, so it sounds like you'd like to retain the old axis time formatting behavior. I've modified my PR to set the `xAxisLabelFormatter`. The example you mentioned looks like:

<img width="775" alt="screen shot 2015-10-10 at 3 04 51 pm" src="https://cloud.githubusercontent.com/assets/132584/10413511/764f3abc-6f60-11e5-9376-5f7ffdc389a0.png">

- Removed changes to axes settings

jjallaire commented:
Subsumed by https://github.com/rstudio/dygraphs/issues/89
